Port dtbook-to-latex script from Pipeline 1

bertfrees opened the pull request:
@egli I have pushed what we worked on. I'm marking it as "draft" because it needs some more changes before we can merge it:

- [x] fix commit message
- [x] add all options
- [x] add author information
- [x] port documentation
- [x] don't hard-code version number in dtbook-to-latex.xsl
- [ ] possibly include volume splitting step
  - moved to a separate issue (#163)
- [ ] possibly copy the images to the result directory
  - moved to a separate issue (#164)
- [x] possibly add some unit tests

Also, we might want to wait for https://github.com/daisy/pipeline1/pull/10 and include those changes here.

Before we merge this, we should also:

- [ ] disable the dtbook-to-latex script provided by Pipeline 1 (currently this means a change to pipeline-framework, although I'm planning to move it to the pipeline-modules repository)

bertfrees committed:
Port dtbook-to-latex script from Pipeline 1

The XSLT code was taken from dtbook2latex_common.xsl in the
daisy/pipeline1 repository at commit 654d9445, with minimal
changes to make it work in XProc:
- Wrap the output in a c:data element so XProc can process it as text
- Update the version comment string

egli reviewed (changes requested):
Generally looks good. It appears though that the pipeline xsl contains some extra fixes that I added over the years. Presumably these need to be added. Ah, I see this is in daisy/pipeline1#10.

bertfrees commented:
How should we divide the list of things to do?

egli committed:
Add all XSL parameters as script options

Expose all parameters from the XSL stylesheet as script options with
correct Pipeline 2 types (enum choices via p:pipeinfo, strings via
px:type="string"). Register the corresponding Datatype OSGi services
in expose-services.

Assisted-by: Claude:claude-sonnet-4-6

egli committed:
Add developer info to pom.xml

Credit Linus Ericson (MTM) for the original Pipeline 1 implementation
and Christian Egli (SBS) for the Pipeline 2 port.

Assisted-by: Claude:claude-sonnet-4-6

egli committed:
Add user documentation for dtbook-to-latex script

Describes the script purpose, lists required LaTeX packages (memoir,
fontspec, babel, etc.), and notes the current limitation around
tagged PDF/PDF UA-2 accessibility support.

Assisted-by: Claude:claude-sonnet-4-6

egli committed:
Inject version string into generated LaTeX at build time

Replace hardcoded version in the XSL with a parameter, pass it from
the XPL using Maven resource filtering (${project.version}). Configure
the build to filter only the XPL file so the XSL is not affected.

Assisted-by: Claude:claude-sonnet-4-6

egli committed:
Add XSpec unit tests for the XSLT stylesheet

Cover minimal DTBook structure, headings (h1-h3), emphasis, strong, and
unordered/ordered lists. A separate XSpec file handles the
replace-em-with-quote scenario, which requires a global x:param
(XSpec 1.0.0 does not support scenario-level parameters).

Assisted-by: Claude:claude-sonnet-4-6

egli commented:
## Copying images to result directory

Leaving this as a follow-up. Here's the reasoning:

The generated `\includegraphics{...}` commands use the relative `@src` paths from the DTBook, so images need to be present at the same relative location under the result directory for LaTeX compilation to work.

However, the current script takes a single DTBook XML file as input (`p:input port="source"`) and therefore has no handle on the accompanying image files. Implementing this properly would require switching to a fileset-based input (like `dtbook-to-epub3` does with `source.fileset` + `source.in-memory`), then copying the non-DTBook entries to the result directory with `px:fileset-store`. That's a non-trivial interface change comparable in scope to the rest of this PR.

The feature is only relevant when `image-visibility` is set to `show` and the user actually wants to compile the .tex file. For now, users who need images will have to copy them manually alongside the generated .tex.

egli commented:
## Volume splitting

The volume splitting step is not ported yet. In Pipeline 1 this was a separate transformer (`ch_sbs_dtbookInsertVolumeSplitPoints`) that inserts `class="volume-split-point"` markers into the DTBook before passing it to `dtbook2latex`.

The current `dtbook-to-latex` script already handles pre-marked input — the XSL has a normalization pass that processes those markers — so no changes to the existing script are needed. The missing piece is porting the split-point insertion step as a separate module.

For Pipeline 2 the natural approach would be to keep the two steps as separate modules (split-point insertion is independent of LaTeX rendering) but expose a combined `dtbook-to-latex-multivolume` script that chains them for the common use case.

Leaving this as a follow-up.

bertfrees commented on scripts/dtbook-to-latex/src/main/resources/xml/dtbook-to-latex.script.xpl:
```diff
@@ -43,6 +43,207 @@ package](http://www.ctan.org/tex-archive/macros/latex/contrib/extsizes).</p>
+					<value>false</value>
+				</choice>
+			</px:type>
+		</p:pipeinfo>
```
This is just a boolean. `choice` results in a dropdown in the GUI, but we just want a checkbox.

bertfrees replied:
(`px:type="boolean"` attribute on `p:option`)

egli replied:
Done — switched to `px:type="boolean"`, dropped the `true`/`false` choice.

bertfrees commented on scripts/dtbook-to-latex/src/main/resources/xml/dtbook-to-latex.script.xpl:
```diff
@@ -43,6 +43,207 @@ package](http://www.ctan.org/tex-archive/macros/latex/contrib/extsizes).</p>
+				</choice>
+			</px:type>
+		</p:pipeinfo>
+	</p:option>
```
This could in theory also be a boolean, if you reword the option. Depends on what you want.

egli replied:
Reworded and renamed to `include-images` (`px:type="boolean"`, default `true`). Wired through the pom.xml service export and the XSLT param as well.

bertfrees commented on scripts/dtbook-to-latex/src/main/resources/xml/dtbook-to-latex.script.xpl:
```diff
@@ -43,6 +43,207 @@ package](http://www.ctan.org/tex-archive/macros/latex/contrib/extsizes).</p>
+	<p:option name="stock-size" required="false" select="'a4paper'">
+		<p:documentation xmlns="http://www.w3.org/1999/xhtml">
+			<h2 px:role="name">Stock size</h2>
+			<p px:role="desc">Stock size for the generated LaTeX.</p>
+		</p:documentation>
+		<p:pipeinfo>
+			<px:type>
+				<choice>
+					<value>a3paper</value>
+					<value>a4paper</value>
+					<value>a5paper</value>
+					<value>letterpaper</value>
+					<value>legalpaper</value>
+					<value>executivepaper</value>
```
Note that you can also attach pretty text to the options, if you would like that.

egli replied:
Added `a:documentation` annotations for each paper size (ISO 216 A3/A4/A5, Letter, Legal, Executive).

bertfrees commented on scripts/dtbook-to-latex/src/main/resources/xml/dtbook-to-latex.script.xpl:
```diff
@@ -43,6 +43,207 @@ package](http://www.ctan.org/tex-archive/macros/latex/contrib/extsizes).</p>
+	<p:option name="replace-em-with-quote" required="false" select="'false'">
+		<p:documentation xmlns="http://www.w3.org/1999/xhtml">
+			<h2 px:role="name">Replace emphasis with quotes</h2>
+			<p px:role="desc">Replace em elements with quoted text (emphasis can be hard to render in large print).</p>
```
Just a comment, I can tweak this myself:

The addition "emphasis can be hard to render in large print" is typically something I would move the second line. The first line of the description is shown in the synopsis of the script in the command line tool. It should be brief if possible, to make it readable on in the terminal.

Note that the name of the option is not shown in the CLI (only the ID and the description), so it is fine if the description repeats parts of the name.

egli replied:
Split the description into two paragraphs — first line is now just "Replace em elements with quoted text.", with the large-print rationale moved to a second paragraph so it doesn't bloat the synopsis.

bertfrees commented on scripts/dtbook-to-latex/src/main/resources/xml/dtbook-to-latex.script.xpl:
```diff
@@ -43,6 +43,207 @@ package](http://www.ctan.org/tex-archive/macros/latex/contrib/extsizes).</p>
+		<p:documentation xmlns="http://www.w3.org/1999/xhtml">
+			<h2 px:role="name">Default language</h2>
+			<p px:role="desc" xml:space="preserve">Language for the babel package when no xml:lang attribute is present.
+
+See the [babel documentation](http://www.ctan.org/get/macros/latex/required/babel/babel.pdf) for all valid values.</p>
+		</p:documentation>
+		<p:pipeinfo>
+			<px:type>
+				<choice>
+					<value>english</value>
+					<value>ngerman</value>
+					<value>USenglish</value>
+					<value>UKenglish</value>
+					<value>swedish</value>
+					<value>canadian</value>
+					<value>french</value>
```
This is a better example of options that would benefit from a nice names in the UI.

egli replied:
Added pretty names for all seven language choices.

bertfrees commented on scripts/dtbook-to-latex/src/main/resources/xml/dtbook-to-latex.script.xpl:
```diff
@@ -43,6 +43,207 @@ package](http://www.ctan.org/tex-archive/macros/latex/contrib/extsizes).</p>
+			<px:type>
+				<choice>
+					<value>english</value>
+					<value>ngerman</value>
```
"ngerman": this seems like a typo?

egli replied:
Not a typo — it's babel's name for the post-1996 reformed German orthography, as opposed to plain `german` (old orthography), which isn't offered here. Added a pretty name ("German (new/reformed orthography, post-1996)") so this is clear in the UI instead of just in a code comment.

bertfrees commented on scripts/dtbook-to-latex/src/test/java/XProcSpecTest.java:
```diff
@@ -0,0 +1,16 @@
+		return new String[] {
+			pipelineModule("file-utils"),
+		};
+	}
```
Just so you know: this is luckily not needed anymore soon. It was annoying that you had to define dependencies that you already have defined in the pom.

bertfrees replied:
I dropped this after rebasing.

bertfrees commented on scripts/dtbook-to-latex/src/main/resources/META-INF/catalog.xml:
```diff
@@ -0,0 +1,15 @@
+  <!--
+      to be able to extend the XSLT in other scripts
+  -->
+  <uri name="http://www.daisy.org/pipeline/modules/dtbook-to-latex/dtbook-to-latex.xsl" uri="../xml/dtbook-to-latex.xsl"/>
```
I'm not sure. Do we actually need this? Or did we add it just in case anyone would like to do it?

egli replied:
Good catch — removed. Nothing in the repo referenced the exposed XSLT, so it was speculative.

bertfrees commented:
Apart from the few comments I made in the code, this looks very good. Thank you!

bertfrees commented on doc/index.md:
```diff
@@ -0,0 +1,15 @@
+
+## Synopsis
+
+{{>synopsis}}
```
This file was added in the wrong place. (My fault.)

egli replied:
Turns out this wasn't fully fixed by moving the real docs to scripts/dtbook-to-latex/doc/index.md — the original stray file at the top-level doc/index.md was still there. Deleted it now.

bertfrees commented:
@egli Have you had a chance to look at my comments? It would be great if we could include this in the next release.

egli commented:
Hi @bertfrees sorry, was on vacation. Will look at it next week

bertfrees commented:
@egli Thanks for addressing all my questions! I don't have more questions for now.
